### PR TITLE
add machine scope

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
## PR Summary

Fixes https://github.com/PowerShell/vscode-powershell/issues/2024

Per the vscode team's request, we now need to mark executables with a `scope: "machine"` to aid in remote development scenarios.

@sandy081 is there a recommended way to test this change? I couldn't use Remote - Container because it just gives me the latest release of the PowerShell extension.

## PR Checklist

Note: Tick the boxes below that apply to this pull request by putting an `x` between the square brackets.
Please mark anything not applicable to this PR `NA`.

- [x] PR has a meaningful title
- [x] Summarized changes
- [ ] N/A PR has tests
- [x] This PR is ready to merge and is not work in progress
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready

### CODE REVIEW

**Comment by sandy081** [comment]
As mentioned, machine scope settings has two characteristics

- They are not overridable at workspace/folder level
- In Remote Set up, User value is ignored.

**Comment by TylerLeonhardt** [comment]
> They are not overridable at workspace/folder level

This is what the `IsExecutable` property does today IIRC

> In Remote Set up, User value is ignored.

Yep perfect for exe paths.

I guess I've done this right, but how do I test it? As in... How can I test a local build of an extension in a remote setup?

File: package.json [hunk 1]
```diff
@@ -755,6 +757,7 @@
         "powershell.developer.powerShellExePath": {
           "type": "string",
           "default": "",
+          "scope": "machine",
           "isExecutable": true,
           "description": "Deprecated. Please use the 'powershell.powerShellExePath' setting instead"
```
Comment (rkeithhill) [inline package.json]: Maybe we should get rid of this setting?

Comment (TylerLeonhardt) [reply package.json]: Agreed

**Comment by TylerLeonhardt** [comment]
well it seems this PR broke our CI... so it did something... I'll switch the test to set a different setting.

**Comment by sandy081** [comment]
@TylerLeonhardt yes, it is - https://code.visualstudio.com/updates/v1_23#_application-scope-settings

**Comment by TylerLeonhardt** [comment]
Ah. Missed that point, thanks! I'll update this PR to remove that.

File: package.json (hunk 1 above)
Comment (TylerLeonhardt) [reply package.json]: done

**Pull request merged**